stl/src/fiopen.cpp

```diff
@@ -58,13 +57,12 @@ FILE* _Xfiopen(const CharT* filename, ios_base::openmode mode, int prot) {
 
     mode &= ~(ios_base::ate | ios_base::_Nocreate | ios_base::_Noreplace);
 
+    // look for a valid mode
     int n = 0;
-    while (valid[n] != 0 && valid[n] != mode) { // look for a valid mode
-        ++n;
-    }
-
-    if (valid[n] == 0) {
-        return nullptr; // no valid mode
+    for (; valid[n] != mode; ++n) {
+        if (n == sizeof(valid) / sizeof(valid[0])) {

```

Comment:
```suggestion
        if (n == size(valid)) {
```
(Don't forget to `#include <iterator>`.)